src/data/nat/factorization.lean

```diff
@@ -93,4 +93,29 @@ end
   factorization (p^k) = single p k :=
 by simp [factorization_pow, hp.factorization]
 
+/-! ### Factorizations of pairs of coprime numbers -/
+
+/-- The prime factorizations of coprime `a` and `b` are disjoint -/
+lemma factorization_disjoint_of_coprime {a b : ℕ} (hab : coprime a b) :
+  disjoint a.factorization.support b.factorization.support :=
+by simpa only [support_factorization]
+  using disjoint_to_finset_iff_disjoint.mpr (coprime_factors_disjoint hab)
+
+/-- For coprime `a` and `b`, the power of `p` in `a * b` is the sum of the powers in `a` and `b` -/
+lemma factorization_mul_of_coprime {a b : ℕ} (hab : coprime a b) :
```

Comment:
This and the below are true without the coprimality assumption, right?

Reply:
Only if we replace that assumption with another condition on `a` and `b` — if `a=0` and `b > 1` then both lemmas fail.

We have `factorization_mul`, which is the counterpart of `factorization_mul_of_coprime` for non-zero `a` and `b`.  But if all we know is that `a` and `b` are coprime then to use that lemma we'd have to fuss about with the separate cases of `a=0, b=1` and vice versa.  So I think it's convenient to have `factorization_mul_of_coprime` as well.

Reply:
Ah, I didn't notice `factorization_mul` was already there, that was what I expected! Thanks for clarifying.

Reply:
Can we have the both nonzero version of `factorization_mul_support_of_coprime` though?